Framework/Algorithms/src/GroupToXResolution.cpp

```diff
@@ -0,0 +1,160 @@
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidAlgorithms/GroupToXResolution.h"
+
+#include "MantidAPI/HistogramValidator.h"
+#include "MantidAPI/WorkspaceHasDxValidator.h"
+#include "MantidDataObjects/Workspace2D.h"
+#include "MantidDataObjects/WorkspaceCreation.h"
+#include "MantidHistogramData/HistogramBuilder.h"
+#include "MantidKernel/BoundedValidator.h"
+#include "MantidKernel/CompositeValidator.h"
+
+#include <boost/math/special_functions/pow.hpp>
+
+namespace {
+namespace Prop {
+std::string const FRACTION{"FractionOfDx"};
+std::string const INPUT_WS{"InputWorkspace"};
+std::string const OUTPUT_WS{"OutputWorkspace"};
+} // namespace Prop
+constexpr double FWHM_GAUSSIAN_EQUIVALENT{0.68};
+} // namespace
+
+namespace Mantid {
+namespace Algorithms {
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(GroupToXResolution)
+
+/// Algorithms name for identification. @see Algorithm::name
+const std::string GroupToXResolution::name() const {
+  return "GroupToXResolution";
+}
+
+/// Algorithm's version for identification. @see Algorithm::version
+int GroupToXResolution::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+const std::string GroupToXResolution::category() const {
+  return "Transforms\\Rebin";
+}
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+const std::string GroupToXResolution::summary() const {
+  return "Groups points within intervals given by the Dx into single points";
+}
+
+/** Initialize the algorithm's properties.
+ */
+void GroupToXResolution::init() {
+  auto inputValidator = boost::make_shared<Kernel::CompositeValidator>();
+  inputValidator->add(boost::make_shared<API::WorkspaceHasDxValidator>());
+  constexpr bool acceptHistograms{false};
+  inputValidator->add(
+      boost::make_shared<API::HistogramValidator>(acceptHistograms));
+  declareProperty(
+      Kernel::make_unique<API::WorkspaceProperty<>>(
+          Prop::INPUT_WS, "", Kernel::Direction::Input, inputValidator),
+      "An input workspace with Dx values.");
+  declareProperty(Kernel::make_unique<API::WorkspaceProperty<>>(
+                      Prop::OUTPUT_WS, "", Kernel::Direction::Output),
+                  "The grouped workspace.");
+  auto betweenZeroAndOne =
+      boost::make_shared<Kernel::BoundedValidator<double>>(0., 1.);
+  betweenZeroAndOne->setLowerExclusive(true);
+  declareProperty(Prop::FRACTION, 0.2, betweenZeroAndOne,
+                  "A fraction of Dx to group the points to.");
+}
+
+std::map<std::string, std::string> GroupToXResolution::validateInputs() {
+  std::map<std::string, std::string> issues;
+  API::MatrixWorkspace_const_sptr inWS = getProperty(Prop::INPUT_WS);
+  if (inWS->getNumberHistograms() != 1) {
+    issues[Prop::INPUT_WS] =
+        "The workspace should contain only a single histogram.";
```

Comment:
Is this not too strict? One can in principle have ragged output.

Reply:
We can expand the algorithm later if the need arises. At this moment, a single histogram treatment suffices for the foreseen use cases.